x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/direct/DirectBlobContainerIndexInput.java

```diff
@@ -190,10 +194,38 @@ private int readFromNewSequentialStream(int part, long pos, byte[] b, int offset
             return 0;
         }
 
+        final long startTimeNanos = directory.statsCurrentTimeNanos();
+
         // if we open a stream of length streamLength then it will not be completely consumed by this read, so it is worthwhile to open
         // it and keep it open for future reads
         final InputStream inputStream = openBlobStream(part, pos, streamLength);
-        streamForSequentialReads = new StreamForSequentialReads(inputStream, part, pos, streamLength);
+        streamForSequentialReads = new StreamForSequentialReads(new FilterInputStream(inputStream) {
+            private int bytesRead = 0;
+
+            @Override
+            public int read() throws IOException {
+                final int result = super.read();
+                if (result != -1) {
+                    bytesRead += result;
+                }
+                return result;
+            }
+
+            @Override
+            public int read(byte[] b, int off, int len) throws IOException {
+                final int result = super.read(b, off, len);
+                if (result != -1) {
+                    bytesRead += result;
+                }
+                return result;
+            }
+
+            @Override
+            public void close() throws IOException {
+                super.close();
+                stats.addOptimizedBytesRead(bytesRead, directory.statsCurrentTimeNanos() - startTimeNanos);
```

Comment:
This also includes the time spent waiting for sequential reads to be requested by the caller. What do you think about timing each `read` call instead and adding all those timings up?

Reply:
You're right. I pushed a940930d46d84d61262dcbde76ab4b984ffd7420 which accumulates numbers for each read operation before adding the sums when the stream is closed. This way the stat will reflect a single read operation that consumed the total bytes of the stream with a more accurate timing.

I adapted the test.